libraries/rush-lib/src/api/RushConfiguration.ts

```diff
@@ -581,11 +354,28 @@ export class RushConfiguration {
     this.__rushPluginsConfiguration = new RushPluginsConfiguration(rushPluginsConfigFilename);
 
     this._npmOptions = new NpmOptionsConfiguration(rushConfigurationJson.npmOptions || {});
-    this._pnpmOptions = new PnpmOptionsConfiguration(
-      rushConfigurationJson.pnpmOptions || {},
-      this._commonTempFolder
-    );
     this._yarnOptions = new YarnOptionsConfiguration(rushConfigurationJson.yarnOptions || {});
+    try {
+      this._pnpmOptions = PnpmOptionsConfiguration.loadFromJsonFileOrThrow(
+        path.join(this._commonRushConfigFolder, RushConstants.pnpmConfigFilename),
```

Comment:
```suggestion
        `${this._commonRushConfigFolder}/${RushConstants.pnpmConfigFilename}`,
```

path.join is somewhat slow, and Windows doesn't care if you use forward or backslashes, so in cases where the parameters are well-known, it's faster to just concatenate the parts of the path with forward slashes.